src/core/thread/network_diagnostic.cpp

```diff
@@ -141,17 +141,20 @@ void NetworkDiagnostic::HandleDiagnosticGetResponse(Coap::Message *         aMes
                                                     const Ip6::MessageInfo *aMessageInfo,
                                                     otError                 aResult)
 {
-    VerifyOrExit(aResult == OT_ERROR_NONE, OT_NOOP);
-    VerifyOrExit(aMessage && aMessage->GetCode() == Coap::kCodeChanged, OT_NOOP);
+    otError error = OT_ERROR_FAILED;
```

Comment:
Similar suggestion here... How about not adding `error` local variable and using `aResult` directly (we can change `aResult` if code is not correct).